src/library_fs.js

```diff
@@ -672,6 +672,31 @@ FS.staticInit();
       }
       return parent.node_ops.mknod(parent, name, mode, dev);
     },
+    statfs(path) {
+
+      // NOTE: None of the defaults here are true. We're just returning safe and
+      //       sane values.
+      var ffree = FS.nextInode - 1; // Free inodes can not be larger than total inodes
+      var defaults = {
+        bsize: 4096,
+        frsize: 4096,
+        blocks: 1e6,
+        bfree: 5e5,
+        bavail: 5e5,
+        files: FS.nextInode,
+        ffree,
+        fsid: 42,
+        flags: 2,
+        namelen: 255,
+      };
+
+      var parent = FS.lookupPath(path, {follow: true}).node;
+      if (typeof parent.node_ops?.statfs !== 'function') {
+        return defaults;
+      }
+
+      return { ...defaults, ...parent.node_ops.statfs(parent.mount.opts.root) };
```

Comment:
How about:

```
var rtn = {
   ... // defaults
}
var parent = FS.lookupPath(path, {follow: true}).node;
if (typeof parent.node_ops?.statfs) {
   Object.assign(rtn, parent.node_ops.statfs(parent.mount.opts.root));
}
return rtn;
```

Also, so we really need to merge the defaults with the actual results?  Are we expecting backends to only paritally fill the out the information?

Reply:
I've applied the suggested changes.

The idea behind merging the defaults it to ensure that all the required fields are always set. In the case of Node, statfs does not return all of the information:

```
StatFs {
  type: 1397114950,
  bsize: 4096,
  blocks: 121938943,
  bfree: 61058895,
  bavail: 61058895,
  files: 999,
  ffree: 1000000
} 
```

We could add the missing fields in the `node_ops.statfs` but it seemed appropriate to have the FS class ensure a valid response. Im happy to change it though, what do you think?